File: pyround1/532.数组中的k-diff数对.py
#
# @lc app=leetcode.cn id=532 lang=python
#
# [532] 数组中的K-diff数对
#

# @lc code=start
class Solution(object):
    def findPairs(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        if k< 0:
            return 0
        if k == 0:
            return len(set([i for i in nums if nums.count(i)>=2]))
        cl = [i + k for i in nums]
        return len(set(cl) & set(nums))
# @lc code=end

File: 532.数组中的-k-diff-数对.cpp
/*
 * @lc app=leetcode.cn id=532 lang=cpp
 *
 * [532] 数组中的 k-diff 数对
 */

// @lc code=start
class Solution {
public:
    int findPairs(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int count = 0;
        for(int i = 0; i <nums.size()-1; i++){
            if(i > 0 and nums[i] == nums[i-1])
                continue;
            for(int j = i+1; j < nums.size(); j++){
                if(j > i+1 and nums[j] == nums[j-1])
                    continue;
                if(abs(nums[j] - nums[i]) == k)
                {
                    count++;
                    break;
                }
            }
        }
        return count;
    }
};
// @lc code=end

